synapse/federation/send_queue.py

```diff
@@ -339,14 +329,6 @@ def federation_ack(self, token):
         for (pos, edu) in edus:
             rows.append((pos, EduRow(edu)))
 
-        # Fetch changed device messages
-        i = self.device_messages.bisect_right(from_token)
-        j = self.device_messages.bisect_right(to_token) + 1
-        device_messages = {v: k for k, v in self.device_messages.items()[i:j]}
-
-        for (destination, pos) in iteritems(device_messages):
-            rows.append((pos, DeviceRow(destination=destination)))
```

Comment:
does this mean that DeviceRow can be removed?

Reply:
Err, yes: #6800